### ISSUE TITLE
google_compute_route doesn't evaluate instance self_link

### ISSUE THREAD

**Issue by seboudry** [terraform-providers/terraform-provider-google#2485]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment
* If an issue is assigned to the "modular-magician" user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If an issue is assigned to a user, that user is claiming responsibility for the issue. If an issue is assigned to "hashibot", a community member has claimed the issue already.

### Terraform Version

```
> terraform -v
Terraform v0.11.10
+ provider.google v1.18.0
```

### Affected Resource(s)

* google_compute_route

### Terraform Configuration Files

```tf
resource "google_compute_route" "route-self-link" {
  name              = "route-self-link"
  network           = "${google_compute_network.cluster.name}"
  tags              = ["k8s"]

  dest_range        = "0.0.0.0/0"

  next_hop_instance = "${google_compute_instance.nat.self_link}"
  priority          = 700
}
```

### Expected Behavior

Field `next_hop_instance` with `${google_compute_instance.nat.self_link}` as plain text evaluated to corresponding instance value.

### Actual Behavior

```
  + google_compute_route.route-self-link
      id:                                                  <computed>
      dest_range:                                          "0.0.0.0/0"
      name:                                                "route-self-link"
      network:                                             "cluster-net"
      next_hop_instance:                                   "${google_compute_instance.nat.self_link}"
      next_hop_network:                                    <computed>
      priority:                                            "700"
      project:                                             <computed>
      self_link:                                           <computed>
      tags.#:                                              "1"
      tags.1699766702:                                     "k8s"
```

### Steps to Reproduce

1. `terraform plan`

### References

I first tough that my problem was linked to the following issue but it was closed
* #677

**Comment by rileykarson** [comment]
Are you getting an error when you run `terraform apply`? At plan time, Terraform doesn't have enough information to know what the self link is, and it will only be able to interpolate it at apply time.

**Comment by rileykarson** [comment]
I'm going to close this out because I think it's expected behaviour, but feel free to reopen if you're encountering an error here.

**Comment by seboudry** [comment]
Hi! Sorry for late response.
Issue solved now (tested with Google provider v1.20.0) even if VM instance not created yet.

Before, I have used a workaround by specifying both instance name (instead of self link) and instance zone:
```
resource "google_compute_route" "route-self-link" {
  name                   = "route-self-link"
  network                = "${google_compute_network.cluster.name}"
  tags                   = ["k8s"]

  dest_range             = "0.0.0.0/0"

  next_hop_instance      = "${google_compute_instance.nat.name}"
  next_hop_instance_zone = "${google_compute_instance.nat.zone}"
  priority               = 700
}
```
